src/plugins/profiling/common/profiling.ts

```diff
@@ -116,15 +129,13 @@ export function compareFrameGroup(a: FrameGroup, b: FrameGroup): number {
   return 0;
 }
 
-export type FrameGroupID = string;
-
 // defaultGroupBy is the "standard" way of grouping frames, by commonly
 // shared group identifiers.
 //
 // For ELF-symbolized frames, group by FunctionName and FileID.
 // For non-symbolized frames, group by FileID and AddressOrLine.
 // Otherwise group by ExeFileName, SourceFilename and FunctionName.
-export function defaultGroupBy(frame: StackFrameMetadata): FrameGroupID {
+export function defaultGroupBy(frame: StackFrameMetadata): FrameGroup {
   const frameGroup = buildFrameGroup();
```

Comment:
Just for readability, I'd suggest a function `newFrameGroup` and rename `buildFrameGroup` into `newFrameGroupFrom`. `newFrameGroup` then calls `newFrameGroupFrom`. I would also make the `partial` parameter non-optional.

Reply:
Thanks, I agree the intent of these methods could have been clearer. I had used `build*` for the method names since this was used in the Go codebase. Based on your feedback, I reviewed the Kibana codebase to find the most common naming conventions for these kind of creation methods.

For example:
* `create[A-Z][a-zA-Z]*` was used 6462 times in Typescript code
* `new[A-Z][a-zA-Z]*` was used 4408 times in Typescript code
* `build[A-Z][a-zA-Z]*` was used 779 times in Typescript code
* `make[A-Z][a-zA-Z]*` was used 348 times in Typescript code

So I decided to use `create*` as the naming convention, especially since `new*` was used for variables and not methods. I also renamed the parameter for the `Partial` value.

However, I kept the optional `Partial` value in the `create*` methods. In the Kibana codebase and the Typescript community, it appears to be common to use creation methods to construct type instances with sensible defaults using an optional `Partial` type.